network_simulator/host.py

```diff
@@ -23,4 +27,65 @@ def __init__(self, address=None, flows={}, flow_packets_received={},
         self.flow_packets_received = flow_packets_received
         self.link = link
 
-    # TODO(team): More event subclasses to come?
\ No newline at end of file
+
+class HostReceivedPacketEvent(Event):
+    """
+    Host receives packets from a flow.
+    """
+
+    def __init__(self, host, packet):
+        """
+        Sanity check: make sure destination was this Host and that this isn't
+        a RouterPacket.
+        :ivar host: host that receives packet
+        :ivar packet: packet received
+        """
+        Event.__init__(self)
+
+        # Sanity check
+        assert packet.dest_id == host and \
+        packet.packet_id != ROUTER_PACKET_DEFAULT_ID
+
+        self.host = host
+        self.packet = packet
+
+    def run(self, main_event_loop, statistics):
+        """
+        Put packet in flow_packets_received and sort.
+        Update statistics.
+        :param main_event_loop: event loop where new Events will be scheduled.
+        :param statistics: statistics to update
+        """
+        # add packet to host.flow_packets_received
+        self.flow_packets_received[self.packet.flow_id] = self.packet.packet_id
+        self.flow_packets_received.sort()
+        
+        statistics.host_packet_received(self.host, self.packet,
+                                        main_event_loop.global_clock_sec)
+
+    def schedule_new_events(self, main_event_loop):
+        """
+        If packet was Ack, look up Flow and say packet was received.
+        Schedule FlowReceivedAckEvent.
+        Populate AckPacket for selective repeat.
```

Comment:
Remove this line. The AckPacket is just a parameter of the FlowReceivedAckEvent constructor; that is all I meant to say in the design doc.